### DISCUSSION TITLE
urlBase / baseurl not working

### DISCUSSION THREAD

**Discussion by mike-vivas** [bpatrik/pigallery2#804] [Q&A]
Hi everyone,

first of all: thanks for creating and sharing this project. This is exactly what I was looking for for sharing my photo collection.

The problem I am facing: "urlBase" settings does not seem to work for me with the docker image. Neither with :latest nor with :edge-alpine images.
I use the docker-compose file, let pigallery2 start and create a new default config.json file. Then stop the container, edit the config.json and put "urlBase": "/test". Then restart the container again.
After starting the container again, the two addresses do not work anymore
- ip:port shows "loading" but gets errors loading js files (see screenshot below)
- ip:port/test shows "Not found"

Anyone else facing this issue? Or am I doing something wrong?

I saw older posts here, saw that the problem was solved in the past or tried out different things that were proposed - but could not get it to work.
I tried to modify this settings in the demo page as well and saw the same behavior.

Main reason behind: I would like to access pigallery2 via "my-domain.com/gallery" instead of "gallery.my-domain.com"

Thanks for any hint,
Best,
Michele

![image](https://github.com/bpatrik/pigallery2/assets/63301828/2bfa3a6f-4351-40bb-9f3c-f2d12d31c97c)

**Comment by VRciF** [comment]
I have the same problem now after upgrading from 1.9.3 to 2.0.0. It seems that within the container the index `/app/dist/en/index.html` is just returned to the browser. Which does not know anything about the `urlBase` configuration.  
@bpatrik Is the `urlBase` feature removed in 2.0.0?

**Comment by bpatrik** [comment]
Please test it on the latest edge

**Reply by VRciF** [reply]
Ah, my fault :see_no_evil: 
It is now working here with the edge docker image (digest `cc9cca7b5144` from 2 days ago).  

The docs (https://github.com/bpatrik/pigallery2/blob/master/docker/README.md) mention the sample config.json [here](https://github.com/bpatrik/pigallery2/blob/master/MANPAGE.md#configjson-sample). I didn't see the updated config beforehand and it seems that `Client` key of the config.json was removed - which I used in 1.9 for urlBase.  

Sorry for my stupidity and thanks for your help and this project!  
Btw: to answer myself how the urlBase works: This is set in the html head as `<base href="<%= Config.Server.urlBase %>/">` :+1:

**Comment by mike-vivas** [comment]
sorry but I cannot confirm - or at least it is not working for me :-(

I pulled the edge-alpine image from 8 hours ago (digest 65af81f93680 in dockerhub) and tried to modify "urlBase" both manually in a freshly created config.json as well as from the UI.
And restarting the docker container in both cases.
In both cases it does not work for me.

When index.html is loaded, I see the `<base href="/test/">` set correctly

But nothing loads and I see the exact same errors as posted in my initial message
